pkg/clean/types.go

```diff
@@ -3,3 +3,27 @@ package clean
 type cleanup interface {
 	Cleanup() error
 }
+
+type CleanupService struct {
+	cleans []cleanup
+	err    error
+}
+
+func NewCleanupService() *CleanupService {
+	return &CleanupService{}
+}
+
+func (c *CleanupService) AddCleanupService(f func() (cleanup, error)) *CleanupService {
+	var clean cleanup
```

Comment:
the parameter looks complex, how about using type to define that ?

Reply:
good